client/my-sites/site-settings/section-general.jsx

```diff
@@ -19,7 +21,7 @@ const SiteSettingsGeneral = ( {
 		<div className="site-settings__main general-settings">
 			<GeneralForm site={ site } />
 
-			{ config.isEnabled( 'manage/site-settings/delete-site' ) && ! site.jetpack && ! site.is_vip &&
+			{ config.isEnabled( 'manage/site-settings/delete-site' ) && site && ! site.jetpack && ! site.is_vip &&
```

Comment:
I'll pass this one; it seems we don't have a selector for VIP yet, but we could add one later on.